src/components/Context.js

```diff
@@ -186,12 +186,10 @@ class ProductProvider extends Component {
         ],
       });
     }
-    //Coupon revalidation logic
+    // Coupon revalidation logic
     if (this.state.appliedVoucher) {
-      // eslint-disable-next-line
-      const revalidate = true;
-      console.log("Voucher applied. Item added - Remove voucher");
-      this.removePromotionFromCart();
+      console.log("Voucher applied. Item added - revalidate voucher");
+      this.addPromotionToCart(this.state.appliedVoucher.code, this.state.appliedVoucher.customer);
```

Comment:
1. would be nice to either `await` it, or at least add a `.catch(logging)`.

2. Also, current implementation will pop up the "voucher applied" toast notification. You might want to suppress it in case of revalidation. If so, I suggest to add a param to `addPromotionToCart(couponCode, customer, suppressNotification = false)`